db/database_collection.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2022-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package db
+
+import (
+	"context"
+
+	"github.com/couchbase/sync_gateway/auth"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbase/sync_gateway/channels"
+)
+
+// DatabaseCollection provides a representation of a single collection of a database.
+type DatabaseCollection struct {
+	Bucket        base.Bucket      // Storage
+	revisionCache RevisionCache    // Cache of recently-accessed doc revisions
```

Comment:
At what point can we remove DatabaseContext.revisionCache?  I'm unsure about adding it here before it's removed there.

Reply:
Yes, removed.